From 1d405353381106c0138bf64cecd62be8518c1dca Mon Sep 17 00:00:00 2001
From: Mohan Kumar G <mkumarg@qti.qualcomm.com>
Date: Thu, 24 Jul 2025 15:39:49 +0530
Subject: [PATCH 1/4] UPSTREAM:PROTOCOL:HOSTAPD: Define structure to hold
 sta_add_params of per-sta profiles

Currently, ML reconfiguration requests from a non-AP MLD are processed
individually, with each link addition request triggering a separate
add_link_sta operation.

Introduce a new structure, ml_reconf_req, to collect sta_add_params for
all STA profiles required for add_link_sta operation.

Patch-work: None

Depends-on: None

Feature: ML_RECONFIG

Signed-off-by: Mohan Kumar G <mkumarg@qti.qualcomm.com>
---
 src/ap/sta_info.h    | 1 +
 src/drivers/driver.h | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/src/ap/sta_info.h b/src/ap/sta_info.h
index c5251bd..1616b14 100644
--- a/src/ap/sta_info.h
+++ b/src/ap/sta_info.h
@@ -339,6 +339,7 @@ struct sta_info {
 	struct mld_info mld_info;
 	u8 mld_assoc_link_id;
 	struct link_reconf_req_list *reconf_req;
+	struct hostapd_sta_add_params *recfg_sta_add_params[MAX_NUM_MLD_LINKS];
 #endif /* CONFIG_IEEE80211BE */
 
 	u16 max_idle_period; /* if nonzero, the granted BSS max idle period in
diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index 4905b04..707c449 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -2974,6 +2974,14 @@ struct hostapd_sta_add_params {
 	size_t eht_240mhz_len;
 };
 
+struct ml_reconf_req {
+	u8 addr[6];
+	struct hostapd_sta_add_params *sta_add_params[MAX_NUM_MLD_LINKS];
+	u16 valid_links;
+	u16 add_links;
+	u16 del_links;
+};
+
 struct mac_address {
 	u8 addr[ETH_ALEN];
 };
-- 
2.34.1

